src/validation.h

```diff
@@ -208,19 +208,22 @@ struct PackageMempoolAcceptResult
 };
 
 /**
- * Try to add a transaction to the mempool. This is an internal function and is
- * exposed only for testing. Client code should use ChainstateManager::ProcessTransaction()
+ * Try to add a transaction to the mempool. This is an internal function and is exposed only for testing.
+ * Client code should use ChainstateManager::ProcessTransaction()
  *
  * @param[in]  active_chainstate  Reference to the active chainstate.
- * @param[in]  pool               Reference to the node's mempool.
```

Comment:
This causes the tx_pool fuzz test to fail since it operates on a mempool that isn't referenced by the `CChainState` object. See https://github.com/bitcoin/bitcoin/pull/23173#discussion_r731895386.

That should probably be fixed in the fuzz test, but it can be left for a future PR. For now, I suggest you just remove the final commit.

Reply:
For reference, this can trivially be fixed by something like:

```cpp
struct MockedChainstate:public CChainstate{
SwapTxPool(CTxMempool*tx_pool){m_mempool=tx_pool;}
}

...

   CTxMempool fuzzed_pool;
   (MockedChainstate*)(&chainstate)->SwapTxPool(&fuzzed_pool);

Reply:
Final commit removed.

Reply:
I also addressed the `tx_pool.cpp` issue using a different approach in #23465.
The fuzz test performed successfully in local machine and CI.

```diff
- const auto res = WITH_LOCK(::cs_main, return AcceptToMemoryPool(chainstate, tx_pool, tx, bypass_limits));
+ CChainState& chainstate{node.chainman->ActiveChainstate()};
+ chainstate.m_mempool = &tx_pool;

+ const auto res = WITH_LOCK(::cs_main, return AcceptToMemoryPool(chainstate, tx, GetTime(), bypass_limits, /* test_accept= */false));
```